packages/lib/src/typeIs.ts

```diff
@@ -9,3 +9,7 @@ export function button(el?: Node | null): el is HTMLButtonElement {
 export function input(el?: Node | null): el is HTMLInputElement {
   return element(el) && el.tagName.toLowerCase() === 'input';
 }
+
+export function array(item?: any): item is Array<any> {
```

Comment:
`Array<unknown>` would be preferable. Here's an [example](https://www.typescriptlang.org/play/#code/KYDwDg9gTgLgBAMwK4DsDGMCWEVwIZRR4CeAgisQBSYzAC2A-AFz4UCULN9cmAznKUIkAPHgoA+OAG8AUHDhRgMJFFxc6PFLxhi0wCAgFDiAbhkBfGTNCRYiVBmy4CRYgFUUAaxQQA7impaRhYxYg4eIJ5+QVdhVG8-FElZeUVlVQjuTC0ddH1DGJIzSysZZHQsHDgAIwJKBAgIFm0obIBzNmk5eR5DShcScioGiDZOlJ7JtBxtOBAWFCQ6auAoOABeREaAbQAGAF0zSfM4YAAbXmAuycmAelu4AAEYXgBaG2AMd8JoOAAVAAWfDgvABECQZwAJjUrpg6JBeLxMNUzsBujc4NMcnNmjBWig2hsthAjj0SpNMH0Bu4vD5-PVGmNrhi4Pdnm8Pl9VlBfoDgaDwVCYTx4RBEcjUeibljZvM4ItlqsiSM9ocpSdzpdmRiZfA5S12srGqT5JZzEA) demonstrating why.

Reply:
Can you help me understand this example better?

```ts
function bar(foo: string) {
    if (arrayAny(foo)) {
        // In the example, this condition would never be hit given the type of the `foo` argument.
        const x: number = foo[0];
    } else {
        // Why isn't this possible? The `foo` argument is typed as a string, so shouldn't this assignment work?
        // @ts-expect-error This should be impossible
        const x: string = foo;
    }

    if (arrayUnknown(foo)) {
        //@ts-expect-error This should be impossible
        const x: number = foo[0];
    } else {
        const x: string = foo;
    }
}
```

Reply:
if we know `foo` is `string`, then `const x: number = foo[0]` should definitely be an error, regardless of the outcome of `array(foo)`, but the `arrayAny` case doesn't produce an error.
since `arrayAny(foo)` causes the inferred type of `foo` to become `Array<any> & string` which implies that:
`foo[0]: (Array<any> & string)[0]`
`foo[0]: Array<any>[0] & string[0]`
`foo[0]: any & string` (string[0] is the character at position 0)
`foo[0]: any`

Reply:
I think I understand – I was thrown off by the condition that would never evaluate to true.

I'll update 😄